src/main/java/org/jabref/gui/actions/ActionHelper.java

```diff
@@ -101,4 +101,8 @@ public static BooleanExpression hasLinkedFileForSelectedEntries(StateManager sta
         return BooleanExpression.booleanExpression(EasyBind.reduce(stateManager.getSelectedEntries(),
                 entries -> entries.anyMatch(entry -> !entry.getFiles().isEmpty())));
     }
+
+    public static BooleanExpression shouldIndexLinkedFiles(PreferencesService preferencesService) {
+        return preferencesService.getFilePreferences().fulltextIndexLinkedFilesProperty();
+    }
```

Comment:
This is weird. A call to a preferences setting should really not sit in the ActionHelper class.

Reply:
I added it to follow the same pattern as `needsDatabase`, but reverted [`f58b1e8` (#11542)](https://github.com/JabRef/jabref/pull/11542/commits/f58b1e8c8f4c722398150c99e0f5e9ae0be947da).